File: lc-exercises/p509.cpp
#include <iostream>

class Solution {
public:
    int fib(int n) {
		if (n == 0) return 0;
		if (n == 1) return 1;

		int dp1 = 0;
		int dp2 = 1;
		int res = 0;

		for (int i = 2; i <= n; i++) {
			res = dp1 + dp2; // 递推关系
			dp1 = dp2;	
			dp2 = res;
		}

		return res;
    }
};

int main() {
	Solution s;
	for (int i = 0; i <= 10; i++) {
		std::cout << s.fib(i) << '\t';
	}

}

File: lc-exercises/p509.py
class Solution(object):
	def fib(self, n):
		"""
		:type n: int
		:rtype: int
		"""
		if n == 0: 
			return 0
		if n == 1:
			return 1
		dp1 = 0
		dp2 = 1
		res = 0
		for i in range(2, n+1):
			res = dp1 + dp2
			dp1 = dp2
			dp2 = res

		return res

  
s = Solution()
for i in range(10):
    print(s.fib(i))